.github/workflows/run-test-cases.yml

```diff
@@ -60,10 +60,13 @@ on:
     
 jobs:
   build-containers:
-    # Build containers only if this is a PR ... otherwise, the containers are pulled ghcr
+    # Build containers only if this is a PR ... otherwise, the containers are pulled ghcr,
+    # but this job cannot be skipped because test-cases needs to run after build-containers
+    # (and if we skip this job, any job that `needs` it will be skipped)
     if: >-
       ( github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.fork == true ) ||
-      ( github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false )
+      ( github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false ) ||
+      ( !(startsWith(github.event_name, 'pull_request')) )
```

Comment:
Can you clarify what falls in this category? Pushes into main?

Reply:
anything that isn't a PR.  so merges (i.e. pushes), releases, and manual triggers.